### PR TITLE
Update/remove vendored packages

### LINKED ISSUE #651: kt-signer does not sign
On a fresh start of the kt-signer:

`CreateEpoch failed: sths.Write(8149281774419562536, 1): signature is nil`

after restarting the signer:

`CreateEpoch failed: sths.Write(8149281774419562536, 2): rpc error: code = DeadlineExceeded desc = context deadline exceeded` 

Might be related to trillian's `sequencer_interval`?
If lowering the sequencing interval is the appropriate way to resolve this we could document this somewhere.
To reproduce, see logs in kubernetes deployment:
```
ktsigner=`kubectl get pods --selector=run=kt-signer -o jsonpath={.items[*].metadata.name}`
kubectl logs $ktsigner
```

### LINKED ISSUE #655: Vendor'ed trillian packages seem out of date
Remove trillian packages from vendor and update dependent vendor packages, if any.

Might be related to #651 #654

### PR DESCRIPTION
**Pull request by Liamsi**
Removed a bunch of outdated vendor packages to make kt compile again with the latest trillian version.

Additional minor changes:
- remove `--verbose` flag / replace with glog's `v` flag
- added info log for the case that no mutations exist 
- adapt to new hasher registry

resolves #655 

also (partly) resolves #651 
(the timeout but not 
`CreateEpoch failed: sths.Write(8924822528332422407, 1): WaitForInclusion(): signature is nil`)

### CODE REVIEW

File: cmd/keytransparency-client/cmd/root.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ import (
 
 	"github.com/google/trillian/client"
 	"github.com/google/trillian/crypto/keys"
-	"github.com/google/trillian/merkle/objhasher"
```
Comment (gdbelvin) [inline cmd/keytransparency-client/cmd/root.go]: Move back to object hasher

File: cmd/keytransparency-signer/Dockerfile [hunk 2]
```diff
@@ -14,6 +14,8 @@ ENV LOG_ID=0 \
 ENV MIN_SIGN_PERIOD=5s \
     MAX_SIGN_PERIOD=24h
 
+ENV VERBOSITY=0
```
Comment (gdbelvin) [inline cmd/keytransparency-signer/Dockerfile]: remove?

File: core/signer/signer.go [hunk 3]
```diff
@@ -254,6 +254,7 @@ func (s *Signer) CreateEpoch(ctx context.Context, forceNewEpoch bool) error {
 	// Don't create epoch if there is nothing to process unless explicitly
 	// specified by caller
 	if len(mutations) == 0 && !forceNewEpoch {
+		glog.Infof("CreateEpoch: no mutations")
```
Comment (gdbelvin) [inline core/signer/signer.go]: CreateEpoch: No mutations. No epoch created

File: cmd/keytransparency-client/cmd/root.go (hunk 1 above)
Comment (Liamsi) [inline cmd/keytransparency-client/cmd/root.go]: Done. PTAL

File: core/signer/signer.go (hunk 3 above)
Comment (Liamsi) [inline core/signer/signer.go]: done

**Pull request merged**